docs/namespace_config_spec.md

```diff
@@ -0,0 +1,309 @@
+# The "Keto Schema Language" Specification
+
+Enforcing fine-grained permissions is a critical building block of mature
+technology solutions that protect privacy and identity in the information age. A
+number of proprietary languages to represent permission already exists, namely
+the [Authzed Schema Language](https://docs.authzed.com/reference/schema-lang)
+and [Auth0 FGA](https://docs.fga.dev/). Many permissions are defined by normal
+developers who typically are most familiar with web technologies like JavaScript
+or Typescript. We therefore see the need for a developer-friendly configuration
+language for permissions that has such a small learning curve that it can be
+picked up, used, and understood with confidence by most developers. We therefore
+chose to define our permissions configuration language as a subset of the most
+common general-purpose programming language: JavaScript/TypeScript.
+
+The Keto Schema Language is a syntactical subset of TypeScript. Along with type
+definitions for the syntax elements of the language (such as `Namespace` or
+`Context`), users can get context help from their IDE while writing the
+configuration.
+
+## Notation
+
+The syntax is specified using Extended Backus-Naur Form (EBNF):
+
+```ebnf
+Production  = production_name "=" [ Expression ] "." .
```

Comment:
Is `Production` common practice wording?

Reply:
Yes, this section is borrowed from the Go language spec.

Reply:
We could just remove this section completely and reference the Go spec. WDYT?

Reply:
In that case I think this is fine :) This probably will never change!